Switch from `dialoguer` to `inquire`

I can't compile vello anymore on 1.76 stable for very strange reasons related to `lazy_static`, see https://github.com/console-rs/dialoguer/issues/306

`dialoguer` is less maintained, less stars, etc. than `inquire`, and dialoguer never made the change to `once_cell` and I think in recent versions of rust (1.75+?) a dep with lifetimes in lazy_static are no longer producing macro-expansion that is valid.

The issue is simple... replace dialoguer with inquire. Should take 5 minutes, but I wanted to explain the purpose here.

We only use `dialoguer` on native to assist with downloads, which we could easily replace with the "Confirm" equivalent on inquire: https://github.com/mikaelmello/inquire?tab=readme-ov-file#Confirm

It happens when compiling `console`, which is a dependant of `dialoguer`. The error seems unhelpful...

```

   Compiling wgpu-types v0.19.2
   Compiling naga v0.19.2
   Compiling rustybuzz v0.12.1
   Compiling read-fonts v0.16.0
   Compiling core-foundation v0.9.4
   Compiling parking_lot_core v0.9.9
   Compiling malloc_buf v0.0.6
   Compiling getrandom v0.2.12
   Compiling memmap2 v0.9.4
   Compiling objc v0.2.7
   Compiling ring v0.17.8
   Compiling fontdb v0.16.2
   Compiling parking_lot v0.12.1
   Compiling errno v0.3.8
   Compiling core-graphics-types v0.1.3
   Compiling metal v0.27.0
   Compiling rustix v0.38.31
   Compiling futures-intrusive v0.5.0
   Compiling rand_core v0.6.4
   Compiling rand_chacha v0.3.1
   Compiling fsevent-sys v4.1.0
   Compiling console v0.15.8
   Compiling filetime v0.2.23
error[E0195]: lifetime parameters or bounds on type `Target` do not match the trait declaration
  --> /Users/simbleau/.cargo/registry/src/index.crates.io-6f17d22bba15001f/console-0.15.8/src/utils.rs:25:1
   |
25 | / lazy_static! {
26 | |     static ref STDOUT_COLORS: AtomicBool = AtomicBool::new(default_colors_enabled(&Term::stdout()));
27 | |     static ref STDERR_COLORS: AtomicBool = AtomicBool::new(default_colors_enabled(&Term::stderr()));
28 | | }
   | |_^ lifetimes do not match type in trait
   |
   = note: this error originates in the macro `__lazy_static_internal` which comes from the expansion of the macro `lazy_static` (in Nightly builds, run with -Z macro-backtrace for more info)

   Compiling rand v0.8.5
error[E0599]: no method named `load` found for struct `STDOUT_COLORS` in the current scope
  --> /Users/simbleau/.cargo/registry/src/index.crates.io-6f17d22bba15001f/console-0.15.8/src/utils.rs:39:19
   |
25 | / lazy_static! {
26 | |     static ref STDOUT_COLORS: AtomicBool = AtomicBool::new(default_colors_enabled(&Term::stdout()));
27 | |     static ref STDERR_COLORS: AtomicBool = AtomicBool::new(default_colors_enabled(&Term::stderr()));
28 | | }
   | |_- method `load` not found for this struct
...
39 |       STDOUT_COLORS.load(Ordering::Relaxed)
   |                     ^^^^ method not found in `STDOUT_COLORS`

error[E0599]: no method named `store` found for struct `STDOUT_COLORS` in the current scope
  --> /Users/simbleau/.cargo/registry/src/index.crates.io-6f17d22bba15001f/console-0.15.8/src/utils.rs:48:19
   |
25 | / lazy_static! {
26 | |     static ref STDOUT_COLORS: AtomicBool = AtomicBool::new(default_colors_enabled(&Term::stdout()));
27 | |     static ref STDERR_COLORS: AtomicBool = AtomicBool::new(default_colors_enabled(&Term::stderr()));
28 | | }
   | |_- method `store` not found for this struct
...
48 |       STDOUT_COLORS.store(val, Ordering::Relaxed)
   |                     ^^^^^ method not found in `STDOUT_COLORS`

error[E0599]: no method named `load` found for struct `STDERR_COLORS` in the current scope
  --> /Users/simbleau/.cargo/registry/src/index.crates.io-6f17d22bba15001f/console-0.15.8/src/utils.rs:60:19
   |
25 | / lazy_static! {
26 | |     static ref STDOUT_COLORS: AtomicBool = AtomicBool::new(default_colors_enabled(&Term::stdout()));
27 | |     static ref STDERR_COLORS: AtomicBool = AtomicBool::new(default_colors_enabled(&Term::stderr()));
28 | | }
   | |_- method `load` not found for this struct
...
60 |       STDERR_COLORS.load(Ordering::Relaxed)
   |                     ^^^^ method not found in `STDERR_COLORS`

error[E0599]: no method named `store` found for struct `STDERR_COLORS` in the current scope
  --> /Users/simbleau/.cargo/registry/src/index.crates.io-6f17d22bba15001f/console-0.15.8/src/utils.rs:69:19
   |
25 | / lazy_static! {
26 | |     static ref STDOUT_COLORS: AtomicBool = AtomicBool::new(default_colors_enabled(&Term::stdout()));
27 | |     static ref STDERR_COLORS: AtomicBool = AtomicBool::new(default_colors_enabled(&Term::stderr()));
28 | | }
   | |_- method `store` not found for this struct
...
69 |       STDERR_COLORS.store(val, Ordering::Relaxed)
   |                     ^^^^^ method not found in `STDERR_COLORS`

Some errors have detailed explanations: E0195, E0599.
For more information about an error, try `rustc --explain E0195`.
error: could not compile `console` (lib) due to 6 previous errors
warning: build failed, waiting for other jobs to finish...
```

--- a/examples/scenes/Cargo.toml
+++ b/examples/scenes/Cargo.toml
@@ -18,7 +18,7 @@ instant = { workspace = true }
 # Used for the `download` command
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies]
 byte-unit = "4.0.19"
-dialoguer = "0.10.4"
+inquire = "0.7"
 ureq = "2.9.6"
 
 [target.'cfg(target_arch = "wasm32")'.dependencies]

--- a/examples/scenes/Cargo.toml
+++ b/examples/scenes/Cargo.toml
@@ -18,7 +18,7 @@ instant = { workspace = true }
 # Used for the `download` command
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies]
 byte-unit = "4.0.19"
-inquire = "0.7"
+inquire = "0.7.0"
 ureq = "2.9.6"
 
 [target.'cfg(target_arch = "wasm32")'.dependencies]

--- a/examples/scenes/src/download.rs
+++ b/examples/scenes/src/download.rs
@@ -9,6 +9,7 @@ use std::{
 use anyhow::{bail, Context, Result};
 use byte_unit::Byte;
 use clap::Args;
+use inquire::Confirm;
 use std::io::Read;
 mod default_downloads;
 
@@ -74,10 +75,7 @@ impl Download {
                     // For rustfmt, split prompt into its own line
                     const PROMPT: &str =
                 "Would you like to download a set of default svg files, as explained above?";
-                    accepted = dialoguer::Confirm::new()
-                        .with_prompt(PROMPT)
-                        .wait_for_newline(true)
-                        .interact()?;
+                    accepted = Confirm::new(PROMPT).with_default(false).prompt()?;
                 } else {
                     println!("Nothing to download! All default downloads already created");
                 }
@@ -101,11 +99,9 @@ impl Download {
                     let cont = if self.auto {
                         false
                     } else {
-                        dialoguer::Confirm::new()
-                            .with_prompt("Would you like to try other downloads?")
-                            .wait_for_newline(true)
-                            .default(false)
-                            .interact()?
+                        Confirm::new("Would you like to try other downloads?")
+                            .with_default(false)
+                            .prompt()?
                     };
                     if !cont {
                         println!("{} downloads complete", completed_count);

--- a/examples/scenes/src/download.rs
+++ b/examples/scenes/src/download.rs
@@ -9,7 +9,6 @@ use std::{
 use anyhow::{bail, Context, Result};
 use byte_unit::Byte;
 use clap::Args;
-use inquire::Confirm;
 use std::io::Read;
 mod default_downloads;
 
@@ -75,7 +74,7 @@ impl Download {
                     // For rustfmt, split prompt into its own line
                     const PROMPT: &str =
                 "Would you like to download a set of default svg files, as explained above?";
-                    accepted = Confirm::new(PROMPT).with_default(false).prompt()?;
+                    accepted = inquire::Confirm::new(PROMPT).with_default(false).prompt()?;
                 } else {
                     println!("Nothing to download! All default downloads already created");
                 }
@@ -99,7 +98,7 @@ impl Download {
                     let cont = if self.auto {
                         false
                     } else {
-                        Confirm::new("Would you like to try other downloads?")
+                        inquire::Confirm::new("Would you like to try other downloads?")
                             .with_default(false)
                             .prompt()?
                     };

